frontend/src/stores/levels/index.ts

```diff
@@ -0,0 +1,22 @@
+import { defineStore } from 'pinia'
+import { getLevels } from './getters'
+import { levels } from './state'
+
+export const useLevelsStore = defineStore('levels', () => {
+  return {
+    levels,
+    getLevels,
+    async fetchLevelsByRoleId(roleId: string) {
+      try {
+        const response = await fetch(`http://localhost:8080/api/levels?role_id=${roleId}`)
```

Comment:
Don't use fetch, use the methods in the http.ts file, that is already wrapping the fetch